SideNav should overlay Content in UIShell

jqlio18 opened the pull request:
Fixes #1463

The expended class should not be applied when the `sideNav` is below the breaking point. I added the `bx--sidenav-fixed` class instead, it seems to works perfectly.

This PR also fixes the hack of `bx--content` that was set previously.

metonym commented on src/UIShell/Content.svelte:
```diff
@@ -11,14 +11,14 @@
   class:bx--content="{true}"
+  style="{unsetLeftMargin ? 'margin-left: 0' : ''}"
   {...$$restProps}
-  style="{unsetLeftMargin ? 'margin-left: 0;' : ''} {$$restProps.style}"
```
I don't think that this should be changed since `$$restProps.style` will override the attribute.

metonym commented:
When testing this locally, the side nav is not appearing when toggling the hamburger menu.

It seems to retain the `bx--side-nav--collapsed` class even when toggled.

Repro

1. `cd docs; yarn dev`
2. Visit `http://127.0.0.1:5173/framed/UIShell/HeaderSwitcher`


https://user-images.githubusercontent.com/10718366/208265955-2e355051-826a-4bca-8038-5615c3fd07d5.mov

jqlio18 commented:
You are right my approach for this problem was wrong. I changed my strategy and took the same approach as React. The main strategy is to take advantage of the `Grid` and it's `offset` propriety to the main content.  I added the same wrappers to the main content as React too

Source : https://v7-react.carbondesignsystem.com/iframe.html?id=components-ui-shell--header-base-w-navigation-actions-and-side-nav&args=&viewMode=story

metonym commented:
Fixed via #2521
